src/zope/app/apidoc/utilities.py

```diff
@@ -41,11 +42,20 @@
 _remove_html_overhead = re.compile(
     r'(?sm)^<html.*<body.*?>\n(.*)</body>\n</html>\n')
 
-space_re = re.compile('\n^( *)\S', re.M)
+space_re = re.compile(r'\n^( *)\S', re.M)
 
 _marker = object()
 
-BASEDIR = dirname(dirname(dirname(dirname(zope.app.__file__))))
+try:
+    _path = zope.app.__file__
+except AttributeError:
+    try:
+        _path = zope.app.__path__[0]
+    except TypeError:
+        # "_NamespacePath" object does not support indexing
+        _path = list(zope.app.__path__)[0]
+BASEDIR = dirname(dirname(dirname(dirname(_path))))
+del _path
 
 def relativizePath(path):
     return path.replace(BASEDIR, 'Zope3')
```

Comment:
I think I might've misunderstood the purpose of this -- if it's meant to shorten the full pathnames for display (and maybe avoid leaking information about filesystem account), it's not that bad -- but not very good either.

It would be better to avoid the BASEDIR nonsense and instead strip the longest matching prefix on sys.path.